### PR TITLE
WebHost: minor css changes to make Supported Games page usable without js

### PR DESCRIPTION
**Pull request by black-sliver**

### CODE REVIEW

**Comment by Silvris** [comment]
Is this different from #2259 ?

**Comment by black-sliver** [comment]
Yes, it's a different approach.

There is even a 3rd possible approach that is probably even cleaner from a DOM PoV: inserting the arrows and assigning the classes with JS, however that has a tendency to be blinky if the javascript is not inline in the html.

And a 4th that is inserting the arrows using css with ::before, hiding the content has the same potential blinkiness issues though.

**Comment by LegendaryLinux** [comment]
I can adjust the JS to not loop over the class elements. Expect a push in a couple days.

**Comment by KonoTyran** [comment]
another alternative would be something like
https://codepen.io/Kono-Tyran/pen/BavqYYB
this uses only css to open/close. then the search feature would just need to toggle checkbox states. page would be fully functional minus search.

**Comment by black-sliver** [comment]
Oh, that's an interesting one.
I personally don't really like the raw HTML it uses (assuming HTML should be data, js and css the representation), and I think "show all" is a valid approach if the user requested a non-interactive (non-js) version of the page, but I'm sure there are uses for it.

**Comment by LegendaryLinux** [comment]
TIL about subsequent combinators. I think @KonoTyran's approach here is ingenious.

If we go the route of Jinja with optional JS everywhere, this sort of thing could preserve interactivity for a lot of our pages when users have JS disabled.

I think in any case, I want to show the JS search/collapse fields, even if they are non-functional for users with JS turned off. I want to include a message like "Hey, look at the cool stuff you could have with JS turned on."

**Comment by Berserker66** [comment]
It's an approach I've used myself on roll20 character sheets, as they don't allow any JS at all. Worked perfectly fine there.

**Comment by black-sliver** [comment]
I don't think promoting "your version" when the user asked for a "no script version" is helpful. Search functionality is still provided by ctrl+f.

Kono's suggestion on the other hand may improve usability of the page by making it faster to read through, however I don't think the old experience was all that bad.

**Comment by LegendaryLinux** [comment]
@black-sliver Let me know what you think of those JS changes. The goal was to remove most id attributes and use relative element referencing.

File: WebHostLib/static/assets/supportedGames.js [hunk 1]
```diff
@@ -1,51 +1,33 @@
...
     if (!evt.target.value.trim()) {
       // If input is empty, display all collapsed games
-      return Array.from(gameHeaders).forEach((header) => {
+      return toggleButtons.forEach((header) => {
         header.style.display = null;
-        const gameName = header.getAttribute('data-game');
-        document.getElementById(`${gameName}-arrow`).innerText = '▶';
-        document.getElementById(gameName).classList.add('collapsed');
+        header.firstElementChild.innerText = '▶';
+        header.classList.add('collapsed');
```
Comment (black-sliver) [inline WebHostLib/static/assets/supportedGames.js:15]:
```suggestion
        header.nextElementSibling.classList.add('collapsed');
```

**Review by black-sliver — COMMENTED** [review]
I think DOM and JS looks a lot cleaner this way.
Found one mistake where it adds the class to the header instead of the content, breaking the collapse-all when clearing the search box.

File: WebHostLib/static/assets/supportedGames.js [hunk 2]
```diff
@@ -1,51 +1,33 @@
...
       // If the game name includes the search string, display the game. If not, hide it
-      if (gameName.toLowerCase().includes(evt.target.value.toLowerCase())) {
+      if (header.getAttribute('data-game').toLowerCase().includes(evt.target.value.toLowerCase())) {
         header.style.display = null;
-        gameArrow.innerText = '▼';
-        gameInfo.classList.remove('collapsed');
+        header.firstElementChild.innerText = '▼';
+        header.nextElementSibling.classList.remove('collapsed');
       } else {
         console.log(header);
```
Comment (black-sliver) [inline WebHostLib/static/assets/supportedGames.js:27]: Also this should probably go?

**Review by black-sliver — COMMENTED** [review]
Not sure if we should leave the `TODO:` or how we want to tackle that in general.  I've never worked with that kind of fragmented js/css where each page has its own set of files.

File: WebHostLib/static/assets/supportedGames.js (hunk 2 above)
Comment (LegendaryLinux) [reply WebHostLib/static/assets/supportedGames.js:27]: lol whoops

**Comment by LegendaryLinux** [comment]
Separate HTML / JS / CSS is how I've always organized things unless a framework is involved. I'm okay with removing the TODO.

**Review by LegendaryLinux — APPROVED** [review]

**Pull request merged**